src/full/Agda/Interaction/Library.hs

```diff
@@ -316,6 +345,67 @@ stripCommentLines = concatMap strip . zip [1..] . lines
     strip (i, s) = [ (i, s') | not $ null s' ]
       where s' = trimLineComment s
 
+-- | Returns the path of the @executables@ file which lists the trusted executables Agda knows about.
+--
+--   Note: file may not exist.
+--
+getExecutablesFile
+  :: IO ExecutablesFile
+getExecutablesFile = do
+  agdaDir <- getAgdaAppDir
+  let defaults = map (agdaDir </>) defaultExecutableFiles  -- NB: non-empty list
+  files <- filterM doesFileExist defaults
+  case files of
+    file : _ -> return $ ExecutablesFile file True
+    []       -> return $ ExecutablesFile (last defaults) False -- doesn't exist, but that's ok
+
+-- | Return the trusted executables Agda knows about.
+--
+--   Returns none if there is no @executables@ file.
+--
+getTrustedExecutables
+  :: LibM (Map ExeName FilePath)  -- ^ Content of @executables@ files.
+getTrustedExecutables = mkLibM [] $ do
+    file <- lift $ getExecutablesFile
+    if not (efExists file) then return Map.empty else do
+      es    <- lift $ stripCommentLines <$> readFile (efPath file)
+      files <- lift $ sequence [ (i, ) <$> expandEnvironmentVariables s | (i, s) <- es ]
```

Comment:
See #4757.  Maybe there are standard solutions for parsing configuration files, expanding variables etc.

Reply:
See [above](https://github.com/agda/agda/pull/4759#discussion_r443252352).

Reply:
Yeah, this was just food for thought from my side, nothing to be addressed in this PR.